File: lesunny/060_스킬트리.java
class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        int[] rank = new int[26];
        
        for(int i=0; i<skill.length(); i++){
            int idx = skill.charAt(i)-'A';
            rank[idx] = i+1;
        }
        
        for(int i=0; i<skill_trees.length; i++){
            int pos = 1;
            boolean flag = true;
            for(int j=0; j<skill_trees[i].length(); j++){
                if(rank[skill_trees[i].charAt(j)-'A']==0){
                    continue;
                }else if(rank[skill_trees[i].charAt(j)-'A']==pos){
                    pos++;
                    continue;
                }else{
                    flag = false;
                }
            }
            if(flag==true)
                answer++;
        }
        
        return answer;
    }
}

File: JIWON/060 스킬트리.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer = 0;
    int rank[27] = {0, };
    
    for(int i=0;i<skill.size();i++){
        rank[skill[i] - 'A'] = i+1;
    }
    
    for(int i=0;i<skill_trees.size();i++){
        int pos = 1;
        bool pass = true;
        for(int j=0;j<skill_trees[i].size();j++){
            if(rank[skill_trees[i][j] -'A'] == 0) continue;
            
            if (rank[skill_trees[i][j] -'A'] == pos){
                pos++;
            } else {
                pass = false;
                break;
            }
        }
        if (pass) answer++;
    }
    
    return answer;
}
